pkg/validate/validate.go

```diff
@@ -93,6 +94,29 @@ func validateProperties(p *api.Properties, externalOnly bool) (errs []error) {
 	}
 	errs = append(errs, validateAgentPoolProfiles(p.AgentPoolProfiles)...)
 	errs = append(errs, validateServicePrincipalProfile(p.ServicePrincipalProfile)...)
+	errs = append(errs, validateAuthProfile(p.AuthProfile)...)
+	return
+}
+
+func validateAuthProfile(ap *api.AuthProfile) (errs []error) {
+	if reflect.DeepEqual(ap, api.AuthProfile{}) {
```

Comment:
shouldn't need reflect here either

Reply:
we dont validate AuthProfiles at all I think because it contains `[]` and with non-pointer interface, it is not necessary, we just validate nested struct?  WDYT?

Reply:
+1